packages/flame/lib/src/components/core/component.dart

```diff
@@ -897,6 +901,8 @@ class Component {
 
 typedef ComponentSetFactory = ComponentSet Function();
 
+enum ChildrenChangedType { add, remove }
```

Comment:
Since this has already happened I think this enum should be in past tense:
```suggestion
enum ChildrenChangedType { added, removed }
```

Reply:
Sure, no problem. I just thought why having past tense twice, since the enum name is already in past tense.

Reply:
I missed that! But I think it is better to have the enum as `ChildrenChangeType` and then the values as past tense, I think it can even be `ChildChangeType` since the change only regards the change of the specific child that we are passing in to the method.

Reply:
Ok, i can switch it.
But i slightly disagree with renaming the enum to ChildChangeType since the change refers to the children list we are interested here. Passing the added or removed child component is just an additional info @st-pasha suggested.

Reply:
Alright, that makes sense, keep it as `Children` then. :)